R/IN22.R

```diff
@@ -0,0 +1,139 @@
+
+#################################################
+# INPUTS FOR THIS SCRIPT - CHANGE THIS SECTION
+
+input_folder = "input/SON25/"
+
+input_file = "2025-01-01-in22-full-dataset.csv"
+
+output_folder_prefix = "../son/content"
+domain = "intermediate_outcomes"
+subdomain = "routes_into_work_(16_to_29_years)"
+indicator_name = "entry_to_higher_education"
+version = "3.0"
+
+indicator_code = "IN22"
+
+
+
+############
+# IMPORTS
+
+source("_shared.R")
+
+
+
+###############
+# THE SCRIPT
+
+input_file_path = paste0(input_folder, input_file)
+
+# Open the input file
+data = read.csv(input_file_path)
+
+# Remove some columns
+data = data_frame__remove_columns(
+  data,
+  "variable_used",
+  "age_used",
+  "weight_used",
+  "year_used",
+  "label"
+)
+
+
+##########################################
+# SECTION: By socio-economic background
+
+section_chart_type = "seb"
+section_csv_name = "SEB"
+
+data_for_section = get_data_for_chart_type(data, section_chart_type)
+
+#################
+# CHART FORMAT
+
+data_for_section = data_frame__sort_rows_with_specific_values(
+  data_frame = data_for_section,
+  column_1 = "secondary_split_value",
+  values_1 = occupational_class_order,
+  # column_2 = "secondary_split_value",
+  # values_2 = occupational_class_order
+) %>% filter(!is.na(secondary_split_value))
+
+
+csv_filename = generate_csv_file_name(split = section_csv_name, format = "chart")
+save_data_frame(data_for_section, csv_filename)
+
+#################
+# TABLE FORMAT
+
+
+pivot_longer_w_ci = data_for_section %>% rename(
+  c("Percent" = "value", "Lower CI" = "lci", "Upper CI" = "uci")) %>% 
+  pivot_longer(cols=c("Percent", "Lower CI", "Upper CI"))
+
+pivot_table = pivot_table__create(
+  pivot_table_source = pivot_longer_w_ci,
+  pivot_columns_column_name = "name",
+  pivot_rows_column_name = "secondary_split_value",
+  pivot_cells_column_name = "value",
+  pivot_table_name = "Socio-economic background",
+  pivot_table_rows_order_values = rev(occupational_class_order),
+  pivot_table_columns_order_values = c("Percent", "Lower CI", "Upper CI"),
+  pivot_table_column_names_suffix = " (%)"
+) 
```

Comment:
* We tend not to include LCI and UCI in our data tables
I think this is because:
  * most/all input files have LCI/UCI, but most pivotted tables (with multiple rows/cols of values) can't support including LCI/UCI for each cell, without the table becoming too large
  * most people don't need to see the LCI/UCI and data analysts can just download the full dataset CSV

* In last year's table, the column was called `Percentage (%)` (rather than `Percent (%)`)

Reply:
- In the tracker "MW: Can the lower and upper confidence intervals for the SEB chart be added to the downloadable chart data CSV", shall I check if the overall download is sufficient?
- renamed

Reply:
> In the tracker "MW: Can the lower and upper confidence intervals for the SEB chart be added to the downloadable chart data CSV"

Ooh, good spot.
Can you check with Matthew?
I think it's important for him to know:
* it will be inconsistent between tables like these (which can have LCI/UCI) and pivotted tables (which can't have LCI/UCI without making the table structure really confusing).
* it'll make the table a bit less easy to read for users (I expect Spence/Alex will want the heading to be "Lower confidence interval (%)" to avoid using an acronym, so the heading will be really long!)

Reply:
I'm waiting for Matthew to ask Rosa and get back to me, but "I think the idea was to have the confidence intervals in the downloadable csv for each chart, but [not] necessarily in the table on the data tab - let me double check with Rosa as its been a while since we discussed it."